### PR TITLE
Remove ES version from security index

### PR DESCRIPTION
**Pull request by jfreden**
This PR removes the use of ES version in the `_meta` field of the mappings for the security system indices (`.security`, `.security-tokens`, `.security-profile`)

To be able to support versionless upgrades of Serverless Elasticsearch, `Version.CURRENT` as an indicator of if something like a system index mapping should be updated or not is no longer sufficient. Instead, the version of a system index needs to be tracked separately from the ES version and bumped if the mapping needs to be updated. 

The `MappingVersion` of the security index is already tracked in the `_meta` information of the mapping, so this PR only switches over to use that and removes the old code. 

We still need to keep the old key around to prevent pre-8.15 and post-8.15 nodes fighting and causing repeated mappings updates in mixed-version clusters. The pre-8.15 nodes will replace the mappings if the `version` field is not present, on the assumption that the mappings are _really_ old.

To reduce use of the `Version.CURRENT` constant this PR changes the `version` field in the security index mappings metadata to be hardcoded as `8.15.0`. Pre-8.15 nodes will see _all_ nodes from 8.15.0 onwards as 8.15.0 for the purposes of determining whether security index mappings need updating. However, that still works as, from the older node's perspective, it means "newer than I understand".

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecuritySystemIndices.java
Comment (gwbrown) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecuritySystemIndices.java:132]: The version meta key is set and immediately re-set here. Admittedly to the correct thing, but still.

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/support/SecurityIndexManagerTests.java
Comment (gwbrown) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/support/SecurityIndexManagerTests.java:428]: I'd prefer it if we could verify that the put mapping call was made a bit more directly, but I'm not going to block this PR if that's difficult.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java
Comment (gwbrown) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java:220]: Could you expand this comment a bit to explain why we want to return the current version in this case (as opposed to `null` or a zero version).

**Review by AthenaEryma — APPROVED** [review]
Left a few minor comments but nothing I feel warrants another round of review, so LGTM once those are addressed to your satisfaction. As I said, I think a lot of this might be basically dead code, but doing enough analysis to rip out the unnecessary parts safely is a bit of a separate project.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecuritySystemIndices.java
Comment (jfreden) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecuritySystemIndices.java:132]: lol yes, setting it once should be enough :) Thanks!

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/support/SecurityIndexManagerTests.java
Comment (jfreden) [reply x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/support/SecurityIndexManagerTests.java]: Yes, I agree. I've updated the test client to use a counter when we get a put mapping request instead. Feels a little better.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java
Comment (jfreden) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java]: Added some more context to the comment. 

If we return a min version that isn't valid here we'll end up trying to get a descriptor compatible with that invalid version (return 0 or null) when we try to update the mapping, so the mapping update fails (the descriptor is null). An option could be to not do an update but I think in this case you want to do an update because the cluster state doesn't any security index mapping version yet and should be updated.

**Pull request merged**
